src/ESPEasyEth.ino

```diff
@@ -0,0 +1,90 @@
+
+#ifdef HAS_ETHERNET
+  // #define ETH_CLK_MODE ETH_CLOCK_GPIO17_OUT
+  // #define ETH_PHY_POWER 12
+  #include "ETH.h"
+  #include <esp_eth.h>
+
+String EthGetHostname()
+{
+  String hostnameToReturn(Settings.getHostname());
+  hostnameToReturn.replace(" ", "-");
+  hostnameToReturn.replace("_", "-"); // See RFC952
+  return hostnameToReturn;
+}
+
+bool checkSettings() {
+  bool result = true;
+  if (Settings.ETH_Phy_Type != 0 && Settings.ETH_Phy_Type != 1)
+    result = false;
+  if (Settings.ETH_Clock_Mode > 3)
+    result = false;
+  if (Settings.ETH_Pin_mdc > MAX_GPIO)
+    result = false;
+  if (Settings.ETH_Pin_mdio > MAX_GPIO)
+    result = false;
+  if (Settings.ETH_Pin_power > MAX_GPIO)
+    result = false;
+  return result;
+}
+
+bool prepareEth() {
+  if (!checkSettings())
+  {
+    addLog(LOG_LEVEL_ERROR, F("ETH: Settings not correct!!!"));
+    return false;
+  }
+  ETH.setHostname(EthGetHostname().c_str());
+  return true;
+}
+
+String getDebugClockModeStr()
+{
+  switch (Settings.ETH_Clock_Mode)
+  {
+    case 0: return F("ETH_CLOCK_GPIO0_IN");
+    case 1: return F("ETH_CLOCK_GPIO0_OUT");
+    case 2: return F("ETH_CLOCK_GPIO16_OUT");
+    case 3: return F("ETH_CLOCK_GPIO17_OUT");
+    default: return F("ETH_CLOCK_ERR");
+  }
+}
+
+void printSettings()
```

Comment:
Same here, it is a global defined function, with a very generic function name.

Reply:
Fixed